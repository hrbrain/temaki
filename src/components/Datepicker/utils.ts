import * as ReactDates from 'react-dates'
import * as Moment from 'moment'

const isInclusivelyBeforeDay = (date: Date) => (day: Moment.Moment) => {
    return ReactDates.isInclusivelyBeforeDay(day, Moment(date))
}

const isInclusivelyAfterDay = (date: Date) => (day: Moment.Moment) => {
    return ReactDates.isInclusivelyAfterDay(day, Moment(date))
}

export const Utils = {
    isInclusivelyAfterDay,
    isInclusivelyBeforeDay
}
