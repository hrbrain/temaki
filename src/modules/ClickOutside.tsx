import * as React from 'react'

type Props = {
    onClickOutside: (e: React.MouseEvent<unknown>) => void
    inactive?: boolean
}
export const Component: React.FC<Props> = props => {
    const divRef = React.useRef<HTMLDivElement>(null)

    const handle = (evt: React.MouseEvent<unknown>) => {
        if (!divRef.current) return
        if (
            !divRef.current.contains(evt.target as Node) &&
            !!props.onClickOutside
        ) {
            props.onClickOutside(evt)
        }
    }

    React.useEffect(() => {
        document.removeEventListener('click', handle as any, true)

        if (props.inactive) return
        document.addEventListener('click', handle as any, true)

        return () => {
            document.removeEventListener('click', handle as any, true)
        }
    }, [props.inactive])

    return <div ref={divRef}>{props.children}</div>
}
