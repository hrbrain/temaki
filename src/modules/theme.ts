import * as React from 'react'
import * as styledComponents from 'styled-components'

export type RequiredThemeColors = {
    primary: {
        default: string
        N20: string
        N40: string
        N60: string
        N80: string
        N95: string
        P20: string
        P40: string
        P60: string
        P80: string
        P95: string
    }
    grayScale: {
        S0: string
        S5: string
        S10: string
        S20: string
        S30: string
        S40: string
        S50: string
        S60: string
        S70: string
        S80: string
        S90: string
        S100: string
    }
    utilities: {
        red: {
            default: string
            N20: string
            N40: string
            N60: string
            N80: string
            N90: string
            N95: string
            P20: string
            P40: string
            P60: string
        }
        blue: string
        highlightGreen: {
            default: string
            N20: string
            N40: string
            N60: string
            N80: string
            N90: string
            N95: string
            P20: string
            P40: string
            P60: string
        }
        paleRed: string
        paleGreen: string
        paleBlue: string
        paleYellow: string
    }
    text: {
        default: string
    }
}

export type RequiredThemeShadows = {
    boxShadow: {
        L1: string
        L2: string
        L3: string
        L4: string
        L5: string
    }
    dropShadow: {
        L1: string
        L2: string
        L3: string
        L4: string
        L5: string
    }
}

export type RequiredThemeTextSize = {
    pc: RequiredThemeTextSizePC
    sp: RequiredThemeTextSizeSP
}

export type RequiredThemeTextSizePC = {
    body: {
        default: string
        L: string
        XL: string
        S: string
    }
    heading: {
        H1: string
        H2: string
        H3: string
        H4: string
        H5: string
        H6: string
    }
}

export type RequiredThemeTextSizeSP = {
    body: {
        default: string
        L: string
        XL: string
        S: string
    }
    heading: {
        H1: string
        H2: string
        H3: string
        H4: string
        H5: string
        H6: string
    }
}

export type RequiredThemeLayers = {
    content: string
    tooltip: string
    toast: string
    header: string
    popover: string
    sidemodal: string
    fullmodal: string
    loading: string
}

export type RequiredThemeProps = {
    colors: RequiredThemeColors
    shadows: RequiredThemeShadows
    typography: RequiredThemeTextSize
    layers: RequiredThemeLayers
}

export const defaultTheme: RequiredThemeProps = {
    colors: {
        primary: {
            default: 'rgb(51, 51, 51)',
            N20: 'rgb(92, 92, 92)',
            N40: 'rgb(133, 133, 133)',
            N60: 'rgb(173, 173, 173)',
            N80: 'rgb(214, 214, 214)',
            N95: 'rgb(245, 245, 245)',
            P20: 'rgb(41, 41, 41)',
            P40: 'rgb(31, 31, 31)',
            P60: 'rgb(20, 20, 20)',
            P80: 'rgb(10, 10, 10)',
            P95: 'rgb(0, 0, 0)'
        },
        grayScale: {
            S0: 'rgb(255, 255, 255)',
            S5: 'rgb(245, 245, 245)',
            S10: 'rgb(234, 234, 234)',
            S20: 'rgb(214, 214, 214)',
            S30: 'rgb(193, 193, 193)',
            S40: 'rgb(173, 173, 173)',
            S50: 'rgb(153, 153, 153)',
            S60: 'rgb(133, 133, 133)',
            S70: 'rgb(112, 112, 112)',
            S80: 'rgb(92, 92, 92)',
            S90: 'rgb(71, 71, 71)',
            S100: 'rgb(51, 51, 51)'
        },
        utilities: {
            red: {
                default: 'rgb(224, 85, 72)',
                N20: 'rgb(230, 119, 109)',
                N40: 'rgb(236, 153, 145)',
                N60: 'rgb(243, 187, 182)',
                N80: 'rgb(249, 221, 218)',
                N90: 'rgb(251, 238, 236)',
                N95: 'rgb(253, 246, 246)',
                P20: 'rgb(179, 68, 57)',
                P40: 'rgb(134, 51, 43)',
                P60: 'rgb(89, 34, 28)'
            },
            blue: 'rgb(48, 120, 191)',
            highlightGreen: {
                default: 'rgb(114, 206, 92)',
                N20: 'rgb(142, 216, 125)',
                N40: 'rgb(170, 226, 157)',
                N60: 'rgb(199, 235, 190)',
                N80: 'rgb(227, 245, 222)',
                N90: 'rgb(240, 250, 238)',
                N95: 'rgb(248, 252, 247)',
                P20: 'rgb(68, 123, 55)',
                P40: 'rgb(91, 164, 73)',
                P60: 'rgb(114, 206, 92)'
            },
            paleRed: 'rgb(253, 246, 246)',
            paleGreen: 'rgb(248 ,252, 247)',
            paleBlue: 'rgb(245, 249, 252)',
            paleYellow: 'rgb(255, 255, 233)'
        },
        text: {
            default: '#333'
        }
    },
    shadows: {
        boxShadow: {
            L1: '0 0 2px 0 rgba(0, 0, 0, 0.16)',
            L2: '0 1px 3px 0 rgba(0, 0, 0, 0.16)',
            L3: '0 2px 5px 0 rgba(0, 0, 0, 0.16)',
            L4: '0 3px 8px 0 rgba(0, 0, 0, 0.16)',
            L5: '0 5px 11px 0 rgba(0, 0, 0, 0.16)'
        },
        dropShadow: {
            L1: '0 0 2px rgba(0, 0, 0, 0.16)',
            L2: '0 1px 3px rgba(0, 0, 0, 0.16)',
            L3: '0 2px 5px rgba(0, 0, 0, 0.16)',
            L4: '0 3px 8px rgba(0, 0, 0, 0.16)',
            L5: '0 5px 11px rgba(0, 0, 0, 0.16)'
        }
    },
    typography: {
        pc: {
            body: {
                XL: '18px',
                L: '16px',
                default: '14px',
                S: '12px'
            },
            heading: {
                H1: '24px',
                H2: '22px',
                H3: '20px',
                H4: '18px',
                H5: '16px',
                H6: '14px'
            }
        },
        sp: {
            body: {
                XL: '18px',
                L: '16px',
                default: '14px',
                S: '12px'
            },
            heading: {
                H1: '24px',
                H2: '22px',
                H3: '20px',
                H4: '18px',
                H5: '16px',
                H6: '14px'
            }
        }
    },
    layers: {
        content: '50',
        tooltip: '100',
        toast: '200',
        header: '300',
        popover: '400',
        sidemodal: '500',
        fullmodal: '600',
        loading: '1000'
    }
}

const casted = styledComponents as styledComponents.ThemedStyledComponentsModule<
    RequiredThemeProps
>

const {
    default: styled,
    ServerStyleSheet,
    StyleSheetManager,
    createGlobalStyle,
    css,
    isStyledComponent,
    keyframes,
    withTheme
} = casted
const ThemeConsumer: React.ExoticComponent<
    React.ConsumerProps<RequiredThemeProps>
> = casted.ThemeConsumer
const ThemeContext: React.Context<RequiredThemeProps> = casted.ThemeContext
const ThemeProvider: React.ComponentClass<
    styledComponents.ThemeProviderProps<RequiredThemeProps, RequiredThemeProps>,
    // tslint:disable-next-line:no-any
    any
> = casted.ThemeProvider

/* eslint-disable import/no-default-export */
export default styled
/* eslint-enable import/no-default-export */
export {
    ServerStyleSheet,
    StyleSheetManager,
    ThemeConsumer,
    ThemeContext,
    ThemeProvider,
    createGlobalStyle,
    css,
    isStyledComponent,
    keyframes,
    withTheme
}
