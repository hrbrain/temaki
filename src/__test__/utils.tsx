import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import * as ReactRouter from 'react-router'
import { defaultTheme, ThemeProvider } from '~/modules/theme'

type WithRouterProps = React.ComponentPropsWithoutRef<typeof ReactRouter.Router>
export const WithRouter: React.FC<WithRouterProps> = props => {
    const { children, ...restProps } = props
    return <ReactRouter.Router {...restProps}>{children}</ReactRouter.Router>
}

// 推論で型エラーがでるので、型定義してキャストする
type ShallowWithTheme = (tree: React.ReactElement) => ShallowWrapper
export const shallowWithTheme: ShallowWithTheme = tree => {
    return shallow(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}

// 推論で型エラーがでるので、型定義してキャストする
type MountWithTheme = (tree: React.ReactElement) => ReactWrapper
export const mountWithTheme: MountWithTheme = tree => {
    return mount(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}
