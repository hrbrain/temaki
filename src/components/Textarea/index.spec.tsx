import * as Enzyme from 'enzyme'
import * as React from 'react'
import { act } from 'react-dom/test-utils'
import { mountWithTheme } from '../../__test__/utils'
import * as Theme from '../../modules/theme'
import 'jest-styled-components'

import * as Textarea from './index'

describe('textarea', () => {
    let wrapper: Enzyme.ReactWrapper
    let mockOnChange: jest.Mock
    let mockOnFocus: jest.Mock
    let mockOnBlur: jest.Mock

    beforeEach(() => {
        mockOnChange = jest.fn()
        mockOnFocus = jest.fn()
        mockOnBlur = jest.fn()
        act(() => {
            wrapper = mountWithTheme(
                <Textarea.Component
                    value={'Testing Value'}
                    onChange={mockOnChange}
                    onFocus={mockOnFocus}
                    onBlur={mockOnBlur}
                />
            )
        })
    })

    it('コンポーネントが存在しているか', () => {
        expect(wrapper.exists()).toBe(true)
    })

    it('以前のスナップショットと一致しているか', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('onChangeが呼ばれたか', () => {
        const textareaEl = wrapper.find('textarea')
        act(() => {
            textareaEl.simulate('change')
            expect(mockOnChange).toHaveBeenCalled()
        })
    })

    it('onFocusが呼ばれたか', () => {
        const textareaEl = wrapper.find('textarea')
        act(() => {
            textareaEl.simulate('focus')
            expect(mockOnFocus).toHaveBeenCalled()
        })
    })

    it('onBlurが呼ばれたか', () => {
        const textareaEl = wrapper.find('textarea')
        act(() => {
            textareaEl.simulate('blur')
            expect(mockOnBlur).toHaveBeenCalled()
        })
    })

    it('onFocus時にCSSが効いているか', () => {
        act(() => {
            expect(wrapper.find('textarea')).toHaveStyleRule(
                'border-color',
                'rgb(114,206,92)',
                {
                    modifier: ':focus'
                }
            )
        })
    })

    it('error時にCSSが効いているか', () => {
        wrapper = mountWithTheme(
            <Textarea.Component
                value={'Testing Value'}
                onChange={mockOnChange}
                onFocus={mockOnFocus}
                onBlur={mockOnBlur}
                errored
            />
        )
        expect(wrapper.find('textarea')).toHaveStyleRule(
            'border-color',
            'rgb(224,85,72)'
        )
    })

    it('errorのメッセージが表示されているか', () => {
        wrapper = mountWithTheme(
            <Textarea.Component
                value={'Testing Value'}
                onChange={mockOnChange}
                onFocus={mockOnFocus}
                onBlur={mockOnBlur}
                errored
                errorMessage="必須項目です"
            />
        )
        const errorMessageEl = wrapper.find('div[data-test="error-message"]')

        expect(errorMessageEl.exists()).toBeTruthy()
        expect(errorMessageEl.text()).toEqual('必須項目です')
    })

    it('diff時にCSSが効いているか', () => {
        wrapper = mountWithTheme(
            <Textarea.Component
                value={'Testing Value'}
                onChange={mockOnChange}
                onFocus={mockOnFocus}
                onBlur={mockOnBlur}
                diff
            />
        )

        expect(wrapper.find('textarea')).toHaveStyleRule(
            'background-color',
            Theme.defaultTheme.colorsOld.utilities.paleYellow.replace(/ /g, '')
        )
    })
})
