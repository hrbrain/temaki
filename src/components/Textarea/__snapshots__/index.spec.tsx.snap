// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea 以前のスナップショットと一致しているか 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: block;
  border: 1px solid rgb(234,234,234);
  border-radius: 6px;
  width: 100%;
  padding: 8px;
  margin-top: 4px;
  font-size: 14px;
  resize: none;
  outline: none;
  color: rgb(51,51,51);
  background-color: inherit;
}

.c1::-webkit-input-placeholder {
  color: rgb(214,214,214);
}

.c1::-moz-placeholder {
  color: rgb(214,214,214);
}

.c1:-ms-input-placeholder {
  color: rgb(214,214,214);
}

.c1::placeholder {
  color: rgb(214,214,214);
}

.c1:focus {
  border-color: rgb(114,206,92);
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "grayScale": Object {
          "S0": "rgb(255, 255, 255)",
          "S10": "rgb(234, 234, 234)",
          "S100": "rgb(51, 51, 51)",
          "S20": "rgb(214, 214, 214)",
          "S30": "rgb(193, 193, 193)",
          "S40": "rgb(173, 173, 173)",
          "S5": "rgb(245, 245, 245)",
          "S50": "rgb(153, 153, 153)",
          "S60": "rgb(133, 133, 133)",
          "S70": "rgb(112, 112, 112)",
          "S80": "rgb(92, 92, 92)",
          "S90": "rgb(71, 71, 71)",
        },
        "primary": Object {
          "N20": "rgb(92, 92, 92)",
          "N40": "rgb(133, 133, 133)",
          "N60": "rgb(173, 173, 173)",
          "N80": "rgb(214, 214, 214)",
          "N95": "rgb(245, 245, 245)",
          "P20": "rgb(41, 41, 41)",
          "P40": "rgb(31, 31, 31)",
          "P60": "rgb(20, 20, 20)",
          "P80": "rgb(10, 10, 10)",
          "P95": "rgb(0, 0, 0)",
          "default": "rgb(51, 51, 51)",
        },
        "text": Object {
          "default": "#333",
        },
        "utilities": Object {
          "blue": "rgb(48, 120, 191)",
          "highlightGreen": Object {
            "N20": "rgb(142, 216, 125)",
            "N40": "rgb(170, 226, 157)",
            "N60": "rgb(199, 235, 190)",
            "N80": "rgb(227, 245, 222)",
            "N90": "rgb(240, 250, 238)",
            "N95": "rgb(248, 252, 247)",
            "P20": "rgb(68, 123, 55)",
            "P40": "rgb(91, 164, 73)",
            "P60": "rgb(114, 206, 92)",
            "default": "rgb(114, 206, 92)",
          },
          "paleBlue": "rgb(245, 249, 252)",
          "paleGreen": "rgb(248 ,252, 247)",
          "paleRed": "rgb(253, 246, 246)",
          "paleYellow": "rgb(255, 255, 233)",
          "red": Object {
            "N20": "rgb(230, 119, 109)",
            "N40": "rgb(236, 153, 145)",
            "N60": "rgb(243, 187, 182)",
            "N80": "rgb(249, 221, 218)",
            "N90": "rgb(251, 238, 236)",
            "N95": "rgb(253, 246, 246)",
            "P20": "rgb(179, 68, 57)",
            "P40": "rgb(134, 51, 43)",
            "P60": "rgb(89, 34, 28)",
            "default": "rgb(224, 85, 72)",
          },
        },
      },
      "shadows": Object {
        "boxShadow": Object {
          "L1": "0 0 2px 0 rgba(0, 0, 0, 0.16)",
          "L2": "0 1px 3px 0 rgba(0, 0, 0, 0.16)",
          "L3": "0 2px 5px 0 rgba(0, 0, 0, 0.16)",
          "L4": "0 3px 8px 0 rgba(0, 0, 0, 0.16)",
          "L5": "0 5px 11px 0 rgba(0, 0, 0, 0.16)",
        },
        "dropShadow": Object {
          "L1": "0 0 2px rgba(0, 0, 0, 0.16)",
          "L2": "0 1px 3px rgba(0, 0, 0, 0.16)",
          "L3": "0 2px 5px rgba(0, 0, 0, 0.16)",
          "L4": "0 3px 8px rgba(0, 0, 0, 0.16)",
          "L5": "0 5px 11px rgba(0, 0, 0, 0.16)",
        },
      },
      "typography": Object {
        "pc": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
        "sp": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
      },
    }
  }
>
  <Textarea
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    value="Testing Value"
  >
    <styled.div>
      <div
        className="c0"
      >
        <Styled(TextareaAutosize)
          onBlur={[MockFunction]}
          onChange={[Function]}
          onFocus={[MockFunction]}
          value="Testing Value"
        >
          <TextareaAutosize
            className="c1"
            inputRef={[Function]}
            onBlur={[MockFunction]}
            onChange={[Function]}
            onFocus={[MockFunction]}
            onHeightChange={[Function]}
            useCacheForDOMMeasurements={false}
            value="Testing Value"
          >
            <textarea
              className="c1"
              onBlur={[MockFunction]}
              onChange={[Function]}
              onFocus={[MockFunction]}
              style={
                Object {
                  "height": 0,
                }
              }
              value="Testing Value"
            />
          </TextareaAutosize>
        </Styled(TextareaAutosize)>
        <FormErrorMessage />
      </div>
    </styled.div>
  </Textarea>
</ThemeProvider>
`;
